src/bundle/ui-dev/src/modules/content-tree/components/content-tree/content.tree.js

```diff
@@ -26,10 +28,52 @@ export default class ContentTree extends Component {
         this.clearDocumentResizingListeners();
     }
 
+    componentDidMount() {
+        this.containerScrollRef.addEventListener('scroll', (event) => {
+            window.clearTimeout(this.scrollTimeout);
+
+            this.scrollTimeout = window.setTimeout(
+                (scrollTop) => {
+                    this.saveConfig('scrollTop', scrollTop);
+                },
+                50,
+                event.currentTarget.scrollTop
+            );
+        });
+    }
+
     componentDidUpdate(prevState) {
         if (this.state.containerWidth !== prevState.containerWidth) {
+            this.saveConfig('width', this.state.containerWidth);
+
             document.body.dispatchEvent(new CustomEvent('ez-content-tree-resized'));
         }
+
+        if (this.props.items && this.props.items.length && !this.scrollPositionRestored) {
+            this.scrollPositionRestored = true;
+
+            this.containerScrollRef.scrollTo(0, this.getConfig('scrollTop'));
+        }
+    }
+
+    saveConfig(id, value) {
+        const data = JSON.parse(window.localStorage.getItem('ez-content-tree-state') || '{}');
+        const userId = window.eZ.helpers.user.getId();
```

Comment:
I don't think it should be taken from window object. To the content tree userId is passed as a prop. It would be better to keep the consistency of data.